File: C++/BitManipulation_Subset_Medium_SubSets_I_(PowerSet)_78.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums)
    {
        vector<vector<int>> matrix;

        for(int i = 0; i < pow(2, nums.size()); i++) //1<<n = 2^n
        {
            vector<int> subset;

            for (int j = 0; j < nums.size(); j++)
            {
                if (i & (1 << j))
                {
                    subset.push_back(nums[j]);
                }
            }
            matrix.push_back(subset);
        }

        return matrix;
    }
};

File: Python/BitManipulation_Subset_Medium_SubSets_I_(PowerSet)_78.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        answer = []
        for i in range(0, ((1<<n))):
            subset = []
            for j in range(0, n):
                if (i & 1<<j) != 0:
                    subset.append(nums[j])
            
            answer.append(subset)
        
        return answer
